agorauikit_android/src/main/java/io/agora/agorauikit_android/AgoraRtmController+Tokens.kt

```diff
@@ -0,0 +1,76 @@
+package io.agora.agorauikit_android
+
+import okhttp3.*
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+import java.util.logging.Level
+import java.util.logging.Logger
+
+public interface RtmTokenCallback {
+    fun onSuccess(token: String)
+    fun onError(error: RtmTokenError)
+}
+
+/**
+ * Error types to expect from fetchToken on failing ot retrieve valid token.
+ */
+enum class RtmTokenError {
+    NO_DATA, INVALID_DATA, INVALID_URL
+}
+
+fun AgoraRtmController.Companion.fetchToken(urlBase: String, rtmId: String, completion: RtmTokenCallback) {
+    val log: Logger = Logger.getLogger("AgoraUIKit")
+    val client = OkHttpClient()
+    val url = "$urlBase/rtm/$rtmId"
+    val request: okhttp3.Request = Request.Builder()
+        .url(url)
+        .method("GET", null)
+        .build()
+
+    try {
+        client.newCall(request).enqueue(object : Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                log.log(Level.WARNING, "Unexpected code ${e.localizedMessage}")
+                completion.onError(RtmTokenError.INVALID_DATA)
+            }
+
+            override fun onResponse(call: Call, response: Response) {
+                response.body()?.string()?.let {
+                    val jObject = JSONObject(it)
+                    val token = jObject.getString("rtmToken")
+                    if (!token.isEmpty()) {
+                        completion.onSuccess(token)
+                        return
+                    }
+                }
+                completion.onError(RtmTokenError.NO_DATA)
+            }
+        })
+//        client.newCall(request).execute().use { response ->
+//            if (!response.isSuccessful()) {
+//                log.log(Level.WARNING, "Unexpected code $response")
+//                completion.onError(RtmTokenError.INVALID_DATA)
+//            } else {
+//                response.body()?.string()?.let {
+//                    val jObject = JSONObject(it)
+//                    val token = jObject.getString("rtmToken")
+//                    if (!token.isEmpty()) {
+//                        completion.onSuccess(token)
+//                        return
+//                    }
+//                }
+//                completion.onError(RtmTokenError.NO_DATA)
+//            }
+//            return
+//        }
+    } catch (e: IOException ) {
+        log.log(Level.WARNING, e.localizedMessage)
+        completion.onError(RtmTokenError.INVALID_URL)
+    } catch (e: JSONException) {
+        log.log(Level.WARNING, e.localizedMessage)
+        completion.onError(RtmTokenError.NO_DATA)
+    } catch (e : Exception) {
+        log.log(Level.WARNING, e.message)
```

Comment:
`completion.onError` should still be called in this exception.
One way would be add another RtmTokenError `UNKNOWN_ERROR` and return that. otherwise completion is not called at all.